src/core/router/net_db/net_db.cc

```diff
@@ -127,46 +131,60 @@ void NetDb::Run() {
+      // builds tunnels for requested destinations
+      if (ts - last_manage_request >=
+          static_cast<std::uint16_t>(NetDbInterval::ManageRequests)) {
         m_Requests.ManageRequests();
         last_manage_request = ts;
       }
-      // save routers, manage leasesets and validate subscriptions every minute
-      if (ts - last_save >= 60) {
+      // save routers, manage leasesets and validate subscriptions
+      if (ts - last_save >= static_cast<std::uint16_t>(NetDbInterval::Save)) {
         if (last_save) {
           SaveUpdated();
           ManageLeaseSets();
         }
         last_save = ts;
       }
-      if (ts - last_publish >= 2400) {  // publish every 40 minutes
+      // publishes router info to a floodfill at Nth interval
+      if (ts - last_publish >=
+	  static_cast<std::uint16_t>(NetDbInterval::PublishRouterInfo)) {
         Publish();
         last_publish = ts;
       }
-      if (ts - last_exploratory >= 30) {  // exploratory every 30 seconds
-        auto num_routers = m_RouterInfos.size();
-        // TODO(anonimal): research these numbers
-        if (num_routers < 2500 || ts - last_exploratory >= 90) {
-          if (num_routers > 0) {
-            num_routers = 800 / num_routers;
-          }
-          if (num_routers < 1)
-            num_routers = 1;
-          if (num_routers > 9)
-            num_routers = 9;
-          m_Requests.ManageRequests();
-          Explore(num_routers);
-          last_exploratory = ts;
+      // builds exploratory tunnels at Nth interval to find more peers
+      if (ts - last_exploratory >= static_cast<std::uint16_t>(NetDbInterval::Exploratory)) {
+        auto known_routers = GetNumRouters();
+        // TODO(olarks): If I'm not mistaken, this should be initialized with
+        // static_cast<std::uint16_t>(NetDbSize::MinExploratoryTunnels) and then
+        // we can lose the else if block entirely (assuming we truly don't care
+        // about delayed exploratory time (see comment below)). I don't think we
+        // ever should have this as 0 - or else we won't explore.
+        std::uint16_t num_routers = 0;
+        // evaluates if a router has a sufficient number of known routers
+        // to use for building tunnels, if less than Nth routers are known,
+        // then more exploratory tunnels will be created to find more routers
+        // for tunnel building
+        if (known_routers > 0
+            && known_routers < static_cast<std::uint16_t>(NetDbSize::MinKnownRouters)) {
+          num_routers = static_cast<std::uint16_t>(NetDbSize::MaxExploratoryTunnels);
+        } else if (
+            (known_routers > static_cast<std::uint16_t>(NetDbSize::MinKnownRouters)
+             && known_routers < static_cast<std::uint16_t>(NetDbSize::FavouredKnownRouters))
+            || ts - last_exploratory >= static_cast<std::uint16_t>(NetDbInterval::DelayedExploratory)) {
+          num_routers = static_cast<std::uint16_t>(NetDbSize::MinExploratoryTunnels);
```

Comment:
```
&anonimal | olark: but you didn't keep the same logic: the condition was originally *within threshold or if delayed, then explore tunnels*. Now, it's *within threshold, always explore tunnels regardless of delay*.          
&anonimal | olark: take a close look at the placement of the explore calls within the original block, I think you may have gotten confused (based on what I saw on earlier formatting changes).
```

This is related to the above TODO. I think both should be resolved before merging.